### ISSUE TITLE
Why image are black?

### ISSUE THREAD

**Issue by throb081** [theEricMa/ScaleDreamer#7]
Hello，i recently run the training code # CUDA_VISIBLE_DEVICES=0  python launch.py \
#     --config configs/multi-prompt_benchmark/asd_mv_triplane_transformer_10k.yaml \
#     --train \
#     system.prompt_processor.prompt_library="instant3d_17000_prompt_library"
however ,when i check the output file,ScaleDreamer/ScaleDreamer-main/outputs/asd_mv_triplane_100k/instant3d_17000_prompt_library@20240819-101117/save/it100000-val/A_distinguished_man_in_a_suit_is_giving_a_gripping_speech_at_a_corporate_conference/,i find that the image-grids are almost black like this : 
![Uploading 屏幕截图 2024-09-02 153048.jpg…]()
Do you know why ?

**Comment by throb081** [comment]
![屏幕截图 2024-09-02 153048](https://github.com/user-attachments/assets/f6f5aa20-90dd-4668-8461-29262fdb2f23)

**Comment by theEricMa** [comment]
Thanks for the question. When everything appears black, it means the training has collapsed, generating empty 3D space. It looks like you're using only one GPU, which might result in an insufficient batch size for stable training. Here are some solutions:

1. Lower the learning rate.
2. Use more GPUs to increase the batch size, or adjust the [gradient accumulation](https://github.com/theEricMa/ScaleDreamer/blob/3942bd20e5eb069ca803648b575c75c91e51fbdc/configs/multi-prompt_benchmark/asd_mv_triplane_transformer_10k.yaml#L129) size for a similar effect.

**Comment by theEricMa** [comment]
You can also check the loss log to ensure the 3D space isn't empty. The `metric.csv` file shows the loss for each iteration, including a term called `train/loss_sparsity`. If this term is around 0.1000, it indicates empty 3D results. Initially, parameters aren't converged, so sparsity will be 0.1000 for a while, but it should rise to between 0.1 and 0.6 within 1000 iterations (using 8 GPUs). Hope this helps!

**Comment by throb081** [comment]
> You can also check the loss log to ensure the 3D space isn't empty. The `metric.csv` file shows the loss for each iteration, including a term called `train/loss_sparsity`. If this term is around 0.1000, it indicates empty 3D results. Initially, parameters aren't converged, so sparsity will be 0.1000 for a while, but it should rise to between 0.1 and 0.6 within 1000 iterations (using 8 GPUs). Hope this helps!

[metrics.csv](https://github.com/user-attachments/files/16844271/metrics.csv)
this is my metrics.csv，it seems to match what you said.loss_sparsity is always 0.1 .If  it indicate empty 3D results ，is there anything i can do to solve it ?

**Comment by theEricMa** [comment]
You can reduce the regularization loss weights. The [sparsity](https://github.com/theEricMa/ScaleDreamer/blob/3942bd20e5eb069ca803648b575c75c91e51fbdc/configs/multi-prompt_benchmark/asd_mv_triplane_transformer_10k.yaml#L104) and [eikonal](https://github.com/theEricMa/ScaleDreamer/blob/3942bd20e5eb069ca803648b575c75c91e51fbdc/configs/multi-prompt_benchmark/asd_mv_triplane_transformer_10k.yaml#L107) losses are regularization terms. However, excessive regularization might cause training to collapse, resulting in empty space output. Try setting `sparsity` loss to 1 or `eikonal` loss to 0.001 to see the difference. Let me know if you have any questions.

**Comment by throb081** [comment]
> You can reduce the regularization loss weights. The [sparsity](https://github.com/theEricMa/ScaleDreamer/blob/3942bd20e5eb069ca803648b575c75c91e51fbdc/configs/multi-prompt_benchmark/asd_mv_triplane_transformer_10k.yaml#L104) and [eikonal](https://github.com/theEricMa/ScaleDreamer/blob/3942bd20e5eb069ca803648b575c75c91e51fbdc/configs/multi-prompt_benchmark/asd_mv_triplane_transformer_10k.yaml#L107) losses are regularization terms. However, excessive regularization might cause training to collapse, resulting in empty space output. Try setting `sparsity` loss to 1 or `eikonal` loss to 0.001 to see the difference. Let me know if you have any questions.

Thank you for your reply。Do you mean change the item lambda_sparsity to 1 and lambda_eikonal to 0.001 at the config file(asd_mv_triplane_transformer_10k.yaml)?
![微信图片_20240903145352](https://github.com/user-attachments/assets/413ffe49-2c6b-4f9b-925c-00ac1cc34bcc)

**Comment by theEricMa** [comment]
Yes, please have a try. 

Since we haven't tested on a single GPU, it's worth finding optimal hyperparameters for that setup. I can allocate one GPU for this experiment and will provide coefficients once it's done.

**Comment by throb081** [comment]
OK,Thankyou very much i will have a try .My device is A6000,i hope what i done can help you too !

**Comment by theEricMa** [comment]
Hi, try setting `lambda_eikonal` to 0.01 while keeping `lambda_sparsity` at 20. I've attached the log for the first few thousand iterations. If you see `loss_sparsity` increasing from 0.1000 upwards, you can expect good results when training is completed.
[metrics.csv](https://github.com/user-attachments/files/16847509/metrics.csv)

**Comment by throb081** [comment]
Did you mean setting lambda_eikonal to 0.001?The original value of lambda_eikonal is 0.01.

**Comment by theEricMa** [comment]
Hi, sorry for the mistake. Please keep `lambda_eikonal` as 0.01 while set `lambda_sparsity` to 5. Attached is my training log.  Please have a try.
[metrics.csv](https://github.com/user-attachments/files/16884918/metrics.csv)

**Comment by theEricMa** [comment]
Hi, I've tested a new configuration for single GPU training. The only changes are:

- lambda_sparsity: 40

- lambda_eikonal: 0.001

The results look promising (see attached image). Let me know if you have any questions.

<img width="460" alt="image" src="https://github.com/user-attachments/assets/e0614755-5b0e-4bc3-a85f-88093394b03a">

**Comment by throb081** [comment]
> Hi, I've tested a new configuration for single GPU training. The only changes are:
> 
> * lambda_sparsity: 40
> * lambda_eikonal: 0.001
> 
> The results look promising (see attached image). Let me know if you have any questions.
> 
> <img alt="image" width="460" src="https://private-user-images.githubusercontent.com/127378729/365909266-e0614755-5b0e-4bc3-a85f-88093394b03a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NTc1NTIsIm5iZiI6MTcyNTk1NzI1MiwicGF0aCI6Ii8xMjczNzg3MjkvMzY1OTA5MjY2LWUwNjE0NzU1LTViMGUtNGJjMy1hODVmLTg4MDkzMzk0YjAzYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMFQwODM0MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MGZiYTE4NzIzMmEwYjRiNTM2NTY1NmMwY2I0NDRkZTYxMjc0MmQyM2VjOWNhMGI5ZDYxNTg3Y2Q5YjU1YzE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.i3plxkIvOZkmKBVi9YTO-T7M-_-bb4BWK0lDqWm1gPs">

Hello,i set the lambda_eikonal: 0.001,and keep  lambda_sparsity: 20 ,i train for about 10w steps .I found that the output images nearly same like this :
![0](https://github.com/user-attachments/assets/60048395-b8ec-49e7-ad2b-81900fc27889)
I find nothing in my metrics.csv,Maybe it's because I interrupted the program。I will try to set  lambda_sparsity: 40 lambda_eikonal: 0.001 and have a try again

**Comment by throb081** [comment]
Hello，i set lambda_sparsity: 40 and lambda_eikonal: 0.001。However，i check the outpus picture and find that they seem like black again...like these :
![屏幕截图 2024-09-15 083827](https://github.com/user-attachments/assets/ab0f2a38-39ae-4f50-8aac-b648a3ce0814)

and here are my csv，it seems like train/loss_sparsity keep 0.1 after 3000steps
[metrics（复件）.csv](https://github.com/user-attachments/files/17004197/metrics.csv)

**Comment by throb081** [comment]
may i have your contact information?here is my QQ:1923388926,for easier communication

**Comment by theEricMa** [comment]
@throb081  and I have developed a more stable configuration for training the Triplane Transformer on a single GPU. We've uploaded this  [config](configs/multi-prompt_benchmark/asd_mv_triplane_transformer_10k_1GPU.yaml) for others to reference.
